src/index.ts

```diff
@@ -3,23 +3,38 @@ import consola from "consola";
 import path from "path";
 import * as cp from "child_process";
 
+/*
+if there are multiple lockfiles in a directory,
+the package manager will get detected in the following order:
+- yarn
+- npm
+- pnpm
+*/
+
+const packageManagers = {
+  yarn: {
+    installCmd: "add",
+    file: "yarn.lock",
+  },
+  npm: {
+    installCmd: "install",
+    file: "package-lock.json",
+  },
+  pnpm: {
+    installCmd: "install",
+    file: "pnpm-lock.yaml",
+  },
+};
+
 export function detectPackageManager(cwd?: string) {
   if (!cwd) cwd = process.cwd();
-  var isYarn = checkForYarn(cwd);
-  var isNpm = checkForNpm(cwd);
-  var isPnpm = checkForPnpm(cwd);
-  // Add pnpm Support
-  if (isNpm) {
-    consola.info("Detected package manager: npm");
-    return "npm";
-  } else if (isYarn) {
-    consola.info("Detected package manager: yarn");
-    return "yarn";
-  } else if (isPnpm) {
-    consola.info("Detected package manager: pnpm");
-    return "pnpm";
+  let pkgMgr = getPkgMgr(cwd);
+  if (pkgMgr != "none") {
+    consola.info(`Detected package manager: ${pkgMgr}`);
+    return pkgMgr;
   } else {
-    consola.error("Can't detect any package manager");
+    consola.warn("Can't detect a package manager, auto fallback to: 'npm'");
```

Comment:
`consola.error`

Reply:
I specifically used warn.  A warning is smarter to indicate something went wrong, but due to the fallback the user can still be sure the package was installed. Throwing an error would suggest the installing failed. The fallback to npm is useful due to many VCS users ignoring the `package-lock.json` file, even when they are using npm as a pkg-manager.

Reply:
Yes, but I wanted to make sure that the fallback is in the interest of the package user. So please add an option to activate or deactivate it.

Reply:
will do